### ISSUE TITLE
[2.0] panic: unaligned 64-bit atomic operation on aarch64/arm64

### ISSUE THREAD

**Issue by skygunner** [owncloud/ocis#6104]
## Describe the bug
server crashed immediately with titled error whenever user upload a file

## Steps to reproduce

start server single binary
login and upload a file

## Actual behavior
```shell
[tusd] 2023/04/21 14:08:15.396793 event="RequestIncoming" method="PATCH" path="/0294d0c0-5a7e-445e-a23f-2cbb5ad5c446" requestId=""
[tusd] 2023/04/21 14:08:15.396976 event="ChunkWriteStart" id="0294d0c0-5a7e-445e-a23f-2cbb5ad5c446" maxSize="445" offset="0"
[tusd] 2023/04/21 14:08:15.397201 event="ChunkWriteComplete" id="0294d0c0-5a7e-445e-a23f-2cbb5ad5c446" bytesWritten="445"
[tusd] 2023/04/21 14:08:15.398063 event="ResponseOutgoing" status="204" method="PATCH" path="/0294d0c0-5a7e-445e-a23f-2cbb5ad5c446" requestId=""
panic: unaligned 64-bit atomic operation

goroutine 1681 [running]:
runtime/internal/atomic.panicUnaligned()
        runtime/internal/atomic/unaligned.go:8 +0x24
runtime/internal/atomic.Xadd64(0xb9821f4, 0x4)
        runtime/internal/atomic/atomic_arm.s:258 +0x14
github.com/blevesearch/zapx/v15.(*chunkedContentCoder).incrementBytesWritten(...)
        github.com/blevesearch/zapx/v15@v15.3.6/contentcoder.go:114
github.com/blevesearch/zapx/v15.(*chunkedContentCoder).flushContents(0xb982180)
        github.com/blevesearch/zapx/v15@v15.3.6/contentcoder.go:143 +0x294
github.com/blevesearch/zapx/v15.(*chunkedContentCoder).Close(...)
        github.com/blevesearch/zapx/v15@v15.3.6/contentcoder.go:110
github.com/blevesearch/zapx/v15.(*interim).writeDicts(0xb7fc000)
        github.com/blevesearch/zapx/v15@v15.3.6/new.go:803 +0x11d8
github.com/blevesearch/zapx/v15.(*interim).convert(0xb7fc000)
        github.com/blevesearch/zapx/v15@v15.3.6/new.go:277 +0x2f8
github.com/blevesearch/zapx/v15.(*ZapPlugin).newWithChunkMode(0x4724798, {0xb7d6f58, 0x1, 0x1}, 0x402)
        github.com/blevesearch/zapx/v15@v15.3.6/new.go:72 +0x218
github.com/blevesearch/zapx/v15.(*ZapPlugin).New(0x4724798, {0xb7d6f58, 0x1, 0x1})
        github.com/blevesearch/zapx/v15@v15.3.6/new.go:47 +0x40
github.com/blevesearch/bleve/v2/index/scorch.(*Scorch).Batch(0x5bc8400, 0xb7c94a0)
        github.com/blevesearch/bleve/v2@v2.3.5/index/scorch/scorch.go:412 +0x5cc
github.com/blevesearch/bleve/v2/index/scorch.(*Scorch).Update(0x5bc8400, {0x33b11d4, 0xb7bdf60})
        github.com/blevesearch/bleve/v2@v2.3.5/index/scorch/scorch.go:335 +0x104
github.com/blevesearch/bleve/v2.(*indexImpl).Index(0x662b130, {0xb7f6690, 0x6e}, {0x1d79548, 0xb7d2d70})
        github.com/blevesearch/bleve/v2@v2.3.5/index_impl.go:258 +0x218
github.com/owncloud/ocis/v2/services/search/pkg/search/index.(*Index).Add(0x631edc8, 0xb7bde60, 0xb58db00)
        github.com/owncloud/ocis/v2/services/search/pkg/search/index/index.go:97 +0x9c
github.com/owncloud/ocis/v2/services/search/pkg/search/provider.(*Provider).doIndexSpace.func1({0x0, 0x0}, 0xb58db00, {0x0, 0x0})
        github.com/owncloud/ocis/v2/services/search/pkg/search/provider/searchprovider.go:337 +0x590
github.com/cs3org/reva/v2/pkg/storage/utils/walker.(*revaWalker).walkRecursively(0xb999e3c, {0x33ac284, 0xb774fc0}, {0x0, 0x0}, 0xb58db00, 0xb999e60)
        github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/walker/walker.go:81 +0x90
github.com/cs3org/reva/v2/pkg/storage/utils/walker.(*revaWalker).Walk(0x643be3c, {0x33ac284, 0xb774fc0}, 0xb908360, 0x643be60)
        github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/walker/walker.go:65 +0x84
github.com/owncloud/ocis/v2/services/search/pkg/search/provider.(*Provider).doIndexSpace(0x662ba90, {0x33ac244, 0x5528000}, 0xb774498, 0xb52ca20)
        github.com/owncloud/ocis/v2/services/search/pkg/search/provider/searchprovider.go:308 +0x6c8
github.com/owncloud/ocis/v2/services/search/pkg/search/provider.New.func1(0xb774498, 0xb52ca20)
        github.com/owncloud/ocis/v2/services/search/pkg/search/provider/searchprovider.go:93 +0x44
github.com/owncloud/ocis/v2/services/search/pkg/search/provider.(*SpaceDebouncer).Debounce.func1()
        github.com/owncloud/ocis/v2/services/search/pkg/search/provider/events.go:60 +0x1a4
created by time.goFunc
        time/sleep.go:176 +0x2c
```

## Setup

Single binary test:
```shell
OCIS_INSECURE=true \
PROXY_HTTP_ADDR=0.0.0.0:9200 \
OCIS_URL=https://oc.test:9200 \
ocis server
```

Linux hostname 5.15.0-1032-oracle #38-Ubuntu SMP Thu Mar 23 19:30:10 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
server cpu is oracle cloud Ampere A1

**Comment by skygunner** [comment]
might be related to 
https://github.com/owncloud/ocis/issues/1887

**Comment by rhafer** [comment]
related upstream bug: https://github.com/blevesearch/zapx/issues/152

it's supposed to be fixed in  zapx@v15.3.9 (part of bleve@v2.3.7). Which we updated to in: https://github.com/owncloud/ocis/pull/5980

**Comment by sparkcloudstudio** [comment]
We're appear to be hitting this issue in oCIS 6.2.0 stable: https://central.owncloud.org/t/panic-unaligned-64-bit-atomic-operation-crash-on-ocis-6-2-0/50326
